### PR TITLE
Request Cache Directives and related code cleanup

### PR DESCRIPTION
**Pull request by dpwspoon**

### CODE REVIEW

**Comment by dpwspoon** [comment]
requires: https://github.com/reaktivity/nukleus-http-cache.spec/pull/29

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/CacheEntry.java [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+        this.responseHeaderSize = responseHeaderSize;
+        this.responseSize = responseSize;
+    }
+
+    private void handleEndOfStream(
+            int msgTypeId,
+            DirectBuffer buffer,
+            int index,
+            int length)
+    {
```
Comment (akrambek) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/CacheEntry.java]: I wonder what is the usage of these signatures if they are not being used.

File: src/test/java/org/reaktivity/nukleus/http_cache/internal/streams/server/ProxyCacheIT.java [hunk 2]
```diff
@@ -210,32 +364,28 @@ public void shouldExpireSMaxage() throws Exception
...
         k3po.finish();
     }
 
     @Test
     @Specification({
         "${route}/proxy/controller",
-        "${streams}/expire.cache.by.default.for.5.seconds/accept/client",
-        "${streams}/expire.cache.by.default.for.5.seconds/connect/server",
+        "${streams}/expire.cache.by.default.for.0.seconds/accept/client",
+        "${streams}/expire.cache.by.default.for.0.seconds/connect/server",
```
Comment (akrambek) [inline src/test/java/org/reaktivity/nukleus/http_cache/internal/streams/server/ProxyCacheIT.java]: I couldn't find this script from spec repo and change name of the test case.

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/CacheControl.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        return values.containsKey(directive);
+    }
+
+    public String getValue(String directive)
+    {
+        return values.get(directive);
+    }
+
+    public List<String> getValues(String directive)
```
[the remark this answers is not in the document]
Comment (stanculescu) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/CacheControl.java]: It seems to me that this method returns a list with all the values of a cache directive:
   Cache-Control: max-age=15, max-age=30 --> returns 15 and 30
but I'm not sure if this is a valid case. Is this the expected behavior?

File: src/test/java/org/reaktivity/nukleus/http_cache/internal/streams/server/ProxyCacheIT.java (hunk 2 above)
Comment (dpwspoon) [reply src/test/java/org/reaktivity/nukleus/http_cache/internal/streams/server/ProxyCacheIT.java]: this is fixed.

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/CacheEntry.java (hunk 1 above)
Comment (dpwspoon) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/CacheEntry.java]: removed.

**Pull request merged**